Add blog for e2e-skeleton plugin

johnSchnake opened the pull request:
Signed-off-by: John Schnake <jschnake@vmware.com>

johnSchnake commented:
Tweaked the ci for the linter; seems odd but extending the timeout did make a meaningful difference. It took 1m28s to complete, just over the 1m default timeout. But really? Because of my one new file?

Regardless, leaving those tweaks in with the blog post.

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+---
+title: "Skip the Boilerplate: A Plugin Ready For Your Business Logic"
```
@johnSchnake I think the title should say something about testing since the example (i.e. making testing easier, or running your test just got easier, or building tests with sonobuoy, etc).

vladimirvivien replied:
Also, maybe put e2e-skeleton (great name) in the title.

johnSchnake replied:
You like the name? I've been on the fence about whether or not we needed a new one. My family is halloween-obsessed so I keep visualizing spooky skeletons :)

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+
+## Introduction
+
+When you decide to test your Kubernetes native code it can be a big chore. When I've been in that position, I do one of the following:
```
`".. When I've been in that position.."` -> What position ? (maybe `When I have to write tests`)

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+   - Simple in concept but you're left with a ton of boilerplate code so that you can do simple things like
+ check if your app is running. You've got to deal with setting up client-go, you've got to figure out all the wiring
+ to set up builds, images, API clients, avoid test collision in the cluster, etc.
+ - vendor upstream k8s testing code
```
Is that even doable? I guess it is, depends on how much pain you are willing to endure when vendoring upstream k8s.

johnSchnake replied:
Yea I believe that you can, you effectively have to vendor _all_ of k/k though and all of its dependencies so its not as useful as you'd think. I never actually ran something that had all that code but I'd seen the LOC for PRs in the hundreds of thousands as a result and it was aborted.

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+   - This has the benefit of giving you lots of helper functions, but it is a huge dependency and bloats your tests. In addition,
+ even with all the new helpers, there is SO much code that it can be confusing about what to use and how to use it.
+
+Instead, we want to suggest you try our new [e2e-skeleton plugin][skeleton]. It is obviously just a starting point (not a plugin you would
```
You have two voices in this writeup. Some places you say `I` and others you say `we`.

johnSchnake replied:
Thanks for pointing it out. I tried to use `we` throughout but thats not the natural one for me so I slipped back and forth I guess.

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+ even with all the new helpers, there is SO much code that it can be confusing about what to use and how to use it.
+
+Instead, we want to suggest you try our new [e2e-skeleton plugin][skeleton]. It is obviously just a starting point (not a plugin you would
+run without modification), but it handles much of the basic boilerplate code for you, including:
```
I would remove ` (not a plugin you would...run without modification)`

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+   - This has the benefit of giving you lots of helper functions, but it is a huge dependency and bloats your tests. In addition,
+ even with all the new helpers, there is SO much code that it can be confusing about what to use and how to use it.
+
+Instead, we want to suggest you try our new [e2e-skeleton plugin][skeleton]. It is obviously just a starting point (not a plugin you would
```
Change  ~`It is obviously`~ > `This plugin provides a starting point for creating and handling...`

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+ - Creates/destroys namespaces for each test to avoid collisions
+ - Sends progress updates to the Sonobuoy aggregator so that `sonobuoy status` can report on test progress while running
+
+In this blog post I'll show off the plugin, its features, and show you how to skip the boilerplate so you can get to testing your business logic.
```
~`show off`~ > `demonstrate the e2e-skeleton plugin...`

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+- Dockerfile
+  - All Sonobuoy plugins end up launching containers in your Kubernetes cluster. So to make a plugin you'll need to containerize your code. We've done that for you already and used a base image that was easy to work with and debug. Feel free to modify this as desired.
+- Instructions/script to build your code
+  - You've got the code and Dockerfile, but you still have to actually build/push the container. We provide a simple `build.sh` script that will build/push the image to your specified registry. (This is currently a very minimal starting point and would love help making it more robust.)
```
`(This is currently a very minimal starting point and would...` > `(This is currently a very minimal starting point and we would...`

vladimirvivien commented on site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md:
```diff
@@ -0,0 +1,135 @@
+
+To get the complete status information, add the `--json` flag.
+
+## Benefits of E2E Framework
```
You should mention above that you are using the e2e-framework for this plugin (at least I didn't see that).

johnSchnake replied:
Oh ok, thanks I'll reread the intro and such and make sure its there. I know I put this section and then mentioned it in the outro but may have lost those bullets when working from my outline.

vladimirvivien reviewed (commented):
@johnSchnake This is a great start, but I think you should get to the point early 
* With the title - This is a tool to write/run custom e2e test for K8s components
* Move this `... plugin skeleton provides a useful jumping off point for starting
to code your own Kubernetes integration tests....` from conclusion to starting paragraphs to establish context early

johnSchnake commented:
Fixed based on your feedback; I did lean on that crutch of using a TL;DR; to get full context early on. Updated some CSS to make block quotes more attractive. Typically used for "NOTE: ..." or "TL;DR; ...."
